### PR TITLE
adds support for the Dataproc on GDC SparkApplication resource

### PR DESCRIPTION
**Pull request by JerryLeiDing**
This PR adds support for the Dataproc on GDC SparkApplication resource to the provider.

```release-note:new-resource
`google_dataproc_gdc_spark_application`
```

**Release Note Template for Downstream PRs (will be copied)**

See [Write release notes](https://googlecloudplatform.github.io/magic-modules/contribute/release-notes/) for guidance.

### CODE REVIEW

**Review by zli82016 — CHANGES_REQUESTED** [review]

**Comment by zli82016** [comment]
Also, can you add update tests for the mutable fields by following the [doc](https://googlecloudplatform.github.io/magic-modules/develop/test/test/#add-an-update-test) ? Thanks.

**Comment by zli82016** [comment]
Is there a Github issue for this PR? If yes, you can link the Github issue in the `description` section of this PR.

File: mmv1/openapi_generate/SparkApplication.yaml [hunk 1]
```diff
@@ -0,0 +1,302 @@
+name: SparkApplication
```
[the remark this answers is not in the document]
Comment (JerryLeiDing) [reply mmv1/openapi_generate/SparkApplication.yaml]: Done

**Comment by JerryLeiDing** [comment]
> Also, can you add update tests for the mutable fields by following the [doc](https://googlecloudplatform.github.io/magic-modules/develop/test/test/#add-an-update-test) ? Thanks.

The resource does not support update currently.

**Comment by zli82016** [comment]
> > Also, can you add update tests for the mutable fields by following the [doc](https://googlecloudplatform.github.io/magic-modules/develop/test/test/#add-an-update-test) ? Thanks.
> 
> The resource does not support update currently.
In this case, can you add `immutable: true` to the resource?

**Comment by JerryLeiDing** [comment]
> > > Also, can you add update tests for the mutable fields by following the [doc](https://googlecloudplatform.github.io/magic-modules/develop/test/test/#add-an-update-test) ? Thanks.
> > 
> > 
> > The resource does not support update currently.
> 
> In this case, can you add `immutable: true` to the resource?

Done

**Comment by JerryLeiDing** [comment]
@zli82016 are there any other changes you would like to see on this PR?

**Review by zli82016 — CHANGES_REQUESTED** [review]

**Comment by zli82016** [comment]
> Hi there, I'm the Modular magician. I've detected the following information about your changes:
> 
> ## Diff report
> Your PR generated some diffs in downstreams - here they are.
> 
> `google` provider: [Diff](https://github.com/modular-magician/terraform-provider-google/compare/auto-pr-12237-old..auto-pr-12237) ( 6 files changed, 2318 insertions(+), 2 deletions(-)) `google-beta` provider: [Diff](https://github.com/modular-magician/terraform-provider-google-beta/compare/auto-pr-12237-old..auto-pr-12237) ( 6 files changed, 2318 insertions(+), 2 deletions(-)) `terraform-google-conversion`: [Diff](https://github.com/modular-magician/terraform-google-conversion/compare/auto-pr-12237-old..auto-pr-12237) ( 1 file changed, 478 insertions(+)) Open in Cloud Shell: [Diff](https://github.com/modular-magician/docs-examples/compare/auto-pr-12237-old..auto-pr-12237) ( 20 files changed, 597 insertions(+))
> 
> ## Missing test report
> Your PR includes resource fields which are not covered by any test.
> 
> Resource: `google_dataproc_gdc_spark_application` (5 total tests) Please add an acceptance test which includes these fields. The test should include the following:
> 
> ```hcl
> resource "google_dataproc_gdc_spark_application" "primary" {
>   spark_sql_application_config {
>     query_file_uri = # value needed
>   }
> }
> ```

Can you add the test for this field `query_file_uri`? Thanks.

File: mmv1/products/dataprocgdc/SparkApplication.yaml [hunk 2]
```diff
@@ -0,0 +1,307 @@
...
+    test_vars_overrides:
+      'project': '"gdce-cluster-monitoring"'
+base_url: projects/{{project}}/locations/{{location}}/serviceInstances/{{serviceinstance}}/sparkApplications
+create_url: projects/{{project}}/locations/{{location}}/serviceInstances/{{serviceinstance}}/sparkApplications?sparkApplicationId={{spark_application_id}}
+self_link: projects/{{project}}/locations/{{location}}/serviceInstances/{{serviceinstance}}/sparkApplications/{{spark_application_id}}
+id_format: projects/{{project}}/locations/{{location}}/serviceInstances/{{serviceinstance}}/sparkApplications/{{spark_application_id}}
+import_format:
+  - projects/{{project}}/locations/{{location}}/serviceInstances/{{serviceinstance}}/sparkApplications/{{spark_application_id}}
+update_verb: PATCH
+update_mask: true
```
[the remark this answers is not in the document]
Comment (JerryLeiDing) [reply mmv1/products/dataprocgdc/SparkApplication.yaml]: Done

**Comment by JerryLeiDing** [comment]
> Can you add the test for this field `query_file_uri`? Thanks.

Done

File: mmv1/templates/terraform/examples/dataprocgdc_sparkapplication_sparksql_query_file.tf.tmpl [hunk 3]
```diff
@@ -0,0 +1,15 @@
+resource "google_dataproc_gdc_spark_application" "{{$.PrimaryResourceId}}" {
+  spark_application_id = "{{index $.Vars "spark_application_id"}}"
+  serviceinstance = "do-not-delete-dataproc-gdc-instance"
+  project         = "{{index $.Vars "project"}}"
+  location        = "us-west2"
+  namespace = "default"
+  display_name = "A SparkSql application for a Terraform create test"
+  spark_sql_application_config {
+    jar_file_uris = ["file:///usr/lib/spark/examples/jars/spark-examples.jar"]
+    main_r_file_uri = "gs://some-bucket/something.sql"
```
Comment (zli82016) [inline mmv1/templates/terraform/examples/dataprocgdc_sparkapplication_sparksql_query_file.tf.tmpl:10]:
```suggestion
    query_file_uri = "gs://some-bucket/something.sql"
```

Comment (JerryLeiDing) [reply mmv1/templates/terraform/examples/dataprocgdc_sparkapplication_sparksql_query_file.tf.tmpl]: Done

**Review by zli82016 — APPROVED** [review]
LGTM

**Pull request merged**